zebra-network/src/peer/connection.rs

```diff
@@ -286,15 +287,21 @@ impl Handler {
                     //
                     // Ignoring the message gives us a chance to synchronize back to the correct
                     // request.
+                    //
+                    // Peers can avoid these cascading errors by sending an explicit `notfound`.
+                    // Zebra sends `notfound`, but `zcashd` doesn't.
```

Comment:
I'm not sure this is related to the cancel request error cascade. If the local request is cancelled, there's no way to notify the remote peer so that it can send us a `notfound` message.

Reply:
Good question!

It's even worse than that, `zcashd` doesn't send `notfound` in response to block requests. (It only sends `notfound` for transactions.)

This is fixed by the `MissingInventoryCollector` in the next PR, specifically:
https://github.com/ZcashFoundation/zebra/pull/3505/files#diff-946777883d52596b10e4679ed99713f02c8afe589c8ac641015b473d82b6dd56R366-R409

Reply:
Oh, I see. I'd suggest rewording this to be clearer:

```suggestion
                    // Peers can avoid these cascading errors in scenario #2 by sending an explicit
                    // `notfound`. Zebra sends `notfound`, but `zcashd` doesn't.
```

Reply:
Expanded in 961edc051, I think it's worth writing a bit more to clarify this scenario, because we know `zcashd` doesn't send `notfound`.